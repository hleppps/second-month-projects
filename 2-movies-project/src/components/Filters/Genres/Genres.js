import React from "react";
import { Button, Form, Spinner } from "react-bootstrap";
import GenreItem from "../GenreItem";

const Genres = ({ allGenresList, selectedGenresList, setGenresFilter }) => {
  const onSelectHandler = (isSelected, genreId) => {
    if (isSelected) {
      setGenresFilter([...selectedGenresList, genreId]);
    }

    if (!isSelected) {
      setGenresFilter(
        selectedGenresList.filter((genreItem) => genreItem !== genreId)
      );
    }
  };

  const resetGenresHandler = () => {
    setGenresFilter([]);
  };

  return (
    <>
      <Button
        data-testid="reset-genres"
        onClick={resetGenresHandler}
        className="my-2 w-100"
        variant="outline-secondary"
      >
        Show All Genres
      </Button>
      <Form.Group className="my-2" data-testid="genres-list">
        {!allGenresList && <Spinner animation="grow" />}
        {allGenresList &&
          allGenresList.map((genreItem) => (
            <GenreItem
              isSelected={selectedGenresList.includes(genreItem.id)}
              onSelect={onSelectHandler}
              key={genreItem.id}
              id={genreItem.id}
              name={genreItem.name}
            />
          ))}
      </Form.Group>
    </>
  );
};

export default Genres;
