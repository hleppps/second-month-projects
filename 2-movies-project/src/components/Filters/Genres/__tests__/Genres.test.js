import { render, cleanup } from "@testing-library/react";
import Genres from "../Genres";

const allGenres = [{ id: 28, name: "Action" }];
const selectedGenresList = [];
const setGenresFilter = jest.fn();

afterEach(cleanup);

describe("Genres components", () => {
  it("renders with content", () => {
    const { getByTestId } = render(
      <Genres
        allGenresList={allGenres}
        selectedGenresList={selectedGenresList}
        setGenresFilter={setGenresFilter}
      />
    );

    expect(getByTestId("genres-list")).not.toBeNull();
  });

  it("matches a snapshot", () => {
    const { asFragment } = render(
      <Genres
        allGenresList={allGenres}
        selectedGenresList={selectedGenresList}
        setGenresFilter={setGenresFilter}
      />
    );
    expect(asFragment(<Genres />)).toMatchSnapshot();
  });
});