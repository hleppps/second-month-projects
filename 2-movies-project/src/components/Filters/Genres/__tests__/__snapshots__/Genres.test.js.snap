// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres components matches a snapshot 1`] = `
<DocumentFragment>
  <button
    class="my-2 w-100 btn btn-outline-secondary"
    data-testid="reset-genres"
    type="button"
  >
    Show All Genres
  </button>
  <div
    class="my-2"
    data-testid="genres-list"
  >
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        data-testid="checkbox"
        id="28"
        type="checkbox"
      />
      <label
        class="form-check-label"
        for="28"
        title=""
      >
        Action
      </label>
    </div>
  </div>
</DocumentFragment>
`;
